ssl/ssl_lib.c

```diff
@@ -5610,7 +5673,7 @@ int SSL_set_record_padding_callback(SSL *ssl,
     BIO *b;
     SSL_CONNECTION *sc = SSL_CONNECTION_FROM_SSL(ssl);
 
-    if (sc == NULL)
+    if (sc == NULL || IS_QUIC(ssl))
```

Comment:
Not sure this is necessary. Its just a no-op for QUIC (the callback will never be called)

Reply:
But the function should fail in this case.

Reply:
SSL_CTX_set_record_padding_callback, SSL_CTX_set_record_padding_callback_arg and SSL_set_record_padding_callback_arg return void so we cannot indicate failure here. I don't see it as inconsistent to allow SSL_set_record_padding_callback_arg as the call has no effect by itself other than accessing a stash variable, and it's harmless if an application wants to continue to use this. Updated SSL_CTX_set_block_padding().